docs/codeql/codeql-cli/about-codeql-packs.rst

```diff
@@ -199,11 +199,14 @@ The ``codeql-pack.lock.yml`` file will contain something like the following:
      my-user/transitive-dependency:
        version: 1.2.4
 
+
 ..
   TODO: Add a link to the CodeQL CLI documentation for query resolution, specifically in regards to resolving from source
 
 The ``codeql/cpp-all`` dependency is locked to version 0.1.4. The ``my-user/my-lib`` dependency is locked to version 0.2.4. The ``my-user/transitive-dependency``, which is a transitive dependency and is not specified in the ``qlpack.yml`` file, is locked to version 1.2.4. The ``other-dependency/from-source`` is absent from the lock file since it is resolved from source. This dependency must be available in the same CodeQL workspace as the pack.
 
+In most cases, the ``codeql-pack.lock.yml`` file is only relevant for query packs since library packs are non-executable and usually do not need their transitive dependencies to be fixed. The exception to this is for library packs that contain tests. In this case, the ``codeql-pack.lock.yml`` file is used to ensure that the tests are always run with the same versions of dependencies to avoid spurious failures when there are mismatched dependencies.
```

Comment:
I'm not 100% sure, but I believe that lock files for library packs may also influence the VS Code QL language server, especially if the repo has only library packs. So, it's not just for tests. That said, I think this paragraph that just mentions tests is fine.

Reply:
On line 26 we say that lock files are relevant for Query packs. It's possible that we should make a minor update to that line since it sounds as if you may get a lock file with all pack types?